src/wire/icmpv4.rs

```diff
@@ -366,6 +366,7 @@ impl<T: AsRef<[u8]>> AsRef<[u8]> for Packet<T> {
 
 /// A high-level representation of an Internet Control Message Protocol version 4 packet header.
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[allow(clippy::manual_non_exhaustive)]
```

Comment:
I'd update these to use  `#[non_exhaustive]`, instead of adding an `#[allow]` for the old way:  #397

Reply:
That was introduced in Rust 1.40 and I didn't want to bump the minimum version.

Reply:
Oh okay, then I don't know what's up with #397 . Could you change it to a crate-wide allow then? Thank you!